File: frontend/imis/lib/roadmap/models/roadmap_gut_check.dart
import 'package:json_annotation/json_annotation.dart';

part 'roadmap_gut_check.g.dart';

@JsonSerializable()
class RoadmapGutCheck {
  final int id;
  final bool isDeleted;
  final String? rowVersion;
  final double ownership;
  final double alignment;
  final double contribution;
  final double measurement;
  final double adaptability;
  final double coherence;
  final double commitment;
  final double totalScore;
  RoadmapGutCheck(
    this.id,
    this.isDeleted,
    this.rowVersion,
    this.ownership,
    this.alignment,
    this.contribution,
    this.measurement,
    this.adaptability,
    this.coherence,
    this.commitment,
    this.totalScore,
  );

  factory RoadmapGutCheck.fromJson(Map<String, dynamic> json) =>
      _$RoadmapGutCheckFromJson(json);

  Map<String, dynamic> toJson() => _$RoadmapGutCheckToJson(this);
}

File: backend/IMIS/IMIS.Domain/RoadmapGutCheck.cs
using Base.Primitives;

namespace IMIS.Domain
{
    public class RoadmapGutCheck :  Entity<int>
    {
        public double Ownership { get; set; }
        public double Alignment { get; set; }
        public double Contribution { get; set; }
        public double Measurement { get; set; }
        public double Adaptability { get; set; }
        public double Coherence { get; set; }
        public double Commitment { get; set; }
        public double TotalScore => (Ownership + Alignment + Contribution + Measurement + Adaptability + Coherence + Commitment) / 7;
        public RoadmapGutCheck() { }
        public RoadmapGutCheck(double score1, double score2, double score3, double score4, double score5, double score6, double score7)
        {
            Ownership = score1;
            Alignment = score2;
            Contribution = score3;
            Measurement = score4;
            Adaptability = score5;
            Coherence = score6; 
            Commitment = score7;
        }
    }
}
